internal/stringset/stringset.go

```diff
@@ -0,0 +1,72 @@
+// Copyright © 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package stringset implements a very basic set collection of strings. Items
+// are unique, and basic set operations of addition, deletion, and existence
+// testing are supported. Additionally, supports exporting the elements as a
+// slice, optionally with the values sorted in ascending order.
+//
+// While the functionality of this package could easily be expanded, it would
+// likely soon be better to use a more full-featured, public set package.
+package stringset
+
+import (
+	"sort"
+)
+
+// Set represents a set collection of strings.
+type Set map[string]struct{}
+
+// New returns a new Set object.
+func New(values ...string) Set {
+	s := make(Set)
+	s.Add(values...)
+	return s
+}
+
+// Add adds one or more strings to the set
+func (s *Set) Add(values ...string) {
+	for _, v := range values {
+		(*s)[v] = struct{}{}
+	}
+}
+
+// Delete removes one or more strings from the set
+func (s *Set) Delete(values ...string) {
+	for _, v := range values {
+		delete(*s, v)
```

Comment:
No error to delete something twice?

Reply:
I considered Contains checks for both addition and deletion, but there wasn't a use case for that yet. Do you think it's important?

Reply:
Since there is no code using this, it is hard to say one way or another!.